bgpd/bgp_fsm.c

```diff
@@ -1442,7 +1442,7 @@ int bgp_start(struct peer *peer)
 	if (!bgp_find_or_add_nexthop(peer->bgp, peer->bgp,
 				     family2afi(peer->su.sa.sa_family), NULL,
 				     peer, connected)) {
-		if (bgp_zebra_num_connects()) {
+		if (!bgp_zebra_num_connects()) {
```

Comment:
I actually think this would break @louberger 's expectation of what should happen here, namely that if you do not have zebra running then it should allow the connection to go through.  We had quite a few arguments about this to get this code to this point and I think this also reverses my expectations of the code in that if we have zebra connected then we expect the connection to be stopped until nht tells us we can get to the peer.

What you are trying to do is setup a 3rd state where we are actually expecting zebra to be here eventually.  I think this is the wrong approach in that you should start zebra first and then bgp if you expect this to work properly.  Additionally there are other cases where zebra if run after bgp starts we will never properly initialize some data structures.  I think I would prefer a different solution here.